aptos-move/framework/table-natives/src/lib.rs

```diff
@@ -510,8 +520,12 @@ fn native_contains_box(
 
     // TODO(Gas): Figure out a way to charge this earlier.
     context.charge(key_cost)?;
-    if let Some(amount) = mem_usage {
-        context.use_heap_memory(amount)?;
+    if !context.timed_feature_enabled(TimedFeatureFlag::FixTableNativesMemoryDoubleCounting)
+        || loaded.is_some()
```

Comment:
This does not save traversals though, can we calculate usage only if flag is on and is some or is none? This can be done above `let mem_usage = ...` and this part stays matching on option, which will be None more often now